src/OpenLoco/src/Ui/WindowManager.cpp

```diff
@@ -823,30 +823,29 @@ namespace OpenLoco::Ui::WindowManager
                 break;
             }
         }
-        if (frontMostWnd != nullptr && frontMostWnd != &_windows[0] && frontMostWnd != w)
+        if (frontMostWnd != nullptr && frontMostWnd != &_windows[0] && frontMostWnd != &w)
         {
-            std::swap(*frontMostWnd, *w);
-            w = frontMostWnd;
-            w->invalidate();
+            std::swap(*frontMostWnd, w);
```

Comment:
All the code after this isn't going to be the same. You've swapped w with front but previously w pointer would then be set to front as that's where w's content moved to. We don't have that change now. It's a bit hard to make this work without pointers tbh. I would probably just make the interface of the function a reference but the body of the function use pointers as before due to this bit causing headaches.

Reply:
Yeah, sorry, I understand the problem.
I can't understand what `invalidate()` does. Does it mark a screen region that should be redrawn?

Reply:
Yes it just marks area of the screen as dirty to force a redraw

Reply:
At this point, to keep working with pointers inside the function, I think I can create the pointer `Window* window = &w;` just before the `if` block and restore `window->` instead of `w.` in subsequent code, returning `window` in the end.